src/samples/audio/smart_amp_llext/CMakeLists.txt

```diff
@@ -53,7 +53,16 @@ target_compile_options(${MODULE} PRIVATE
 	-save-temps -O2
 )
 
+if("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "zephyr")
 set(MODULE_LINKER_PARAMS -nostdlib -nodefaultlibs)
+set(EXTRA_LINKED_PARAMS -shared)
+set(COPY_CMD ${CMAKE_STRIP} -R .xt.* -o ${MODULE}_out.so ${MODULE}_llext.so)
+else()
+set(MODULE_LINKER_PARAMS -nostdlib -nodefaultlibs -r)
+set(EXTRA_LINKED_PARAMS)
+set(COPY_CMD ${CMAKE_OBJCOPY} -R .xt.* ${MODULE}_llext.so ${MODULE}_out.so)
+endif()
+
```

Comment:
you should split code, script changes. Not sure why they belong in the same patch?

Also there's no reason why smart_amp should be included here, this is a separate addition IMHO.

Reply:
smart-amp is currently the only component in "main," that can be built as a module. So while adding generic relocatable format support to SOF, we also add it to smart-amp, but this can be split, yes